### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by naterush**
# Description

This PR moves the Pivot taskpane to use the useLiveUpdatingParams hook. 
In doing so, this leads to some refactors to the useLiveUpdatingParams 
hook that make it more flexible, if a bit more complex. We'll keep an eye
on that balance. 

This PR also makes it so API calls are logged properly. The two cool things
this will give us is:
1. Timing information on API calls (we don't know what is slow!)
2. Information about which API calls fail (we don't know what is fragile!)

There's some weirdness where we dont' actually get the error in threads, 
I am not sure why. But I messed about with it and can't figure out a solution,
so for now this is fine - we can improve in the future! Opened an issue here 
#https://github.com/mito-ds/monorepo/issues/386

# Testing

Test pivot and editing pivot. Test logging of API calls.

# Documentation

No.

### CODE REVIEW

File: mitosheet/mitosheet/api/api.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 # NOTE: BE CAREFUL WITH THIS. When in development mode, you can set it to False
 # so the API calls are handled in the main thread, to make printing easy
-THREADED = True
+THREADED = False
```
Comment (naterush) [inline mitosheet/mitosheet/api/api.py:31]: Change

File: mitosheet/mitosheet/telemetry/telemetry_utils.py [hunk 2]
```diff
@@ -36,7 +36,7 @@
                             is_local_deployment, is_running_test)
 
 # If you want, you can optionally choose to print logs
-PRINT_LOGS = False
+PRINT_LOGS = True
```
Comment (naterush) [inline mitosheet/mitosheet/telemetry/telemetry_utils.py:39]: Change

File: mitosheet/src/components/taskpanes/PivotTable/pivotUtils.tsx [hunk 3]
```diff
@@ -1,30 +1,50 @@
...
         pivotRowColumnIDs: pivotParams.pivot_rows_column_ids,
         pivotColumnsColumnIDs: pivotParams.pivot_columns_column_ids,
         pivotValuesColumnIDsArray: valuesRecordToArray(pivotParams.values_column_ids_map),
-        flattenColumnHeaders: pivotParams.flatten_column_headers
+        flattenColumnHeaders: pivotParams.flatten_column_headers,
+        destinationSheetIndex: pivotParams.destination_sheet_index
     };
 }
 
+export const getPivotFrontendParamsFromBackendParams = (params: FrontendPivotParams): BackendPivotParams => {
```
Comment (naterush) [inline mitosheet/src/components/taskpanes/PivotTable/pivotUtils.tsx:35]: Functions are named wrong!

File: mitosheet/src/hooks/useLiveUpdatingParams.tsx [hunk 4]
```diff
@@ -24,25 +24,39 @@ import { useEffectOnUpdateEvent } from "./useEffectOnUpdateEvent";
     the parameters to the user and allow them to edit them. This is really
     sweet, and we'll continue to migrate to this hook over time. Woo!
 */
-function useLiveUpdatingParams<ParamType>(
-    defaultParams: ParamType | undefined | (() => ParamType | undefined),
+function useLiveUpdatingParams<FrontendParamType, BackendParamType>(
```
Comment (naterush) [inline mitosheet/src/hooks/useLiveUpdatingParams.tsx:27]: TODO: leave a comment explaining this difference!

File: mitosheet/src/jupyter/api.tsx [hunk 5]
```diff
@@ -510,32 +510,6 @@ export default class MitoAPI {
         return undefined;
     }
```
Comment (naterush) [inline mitosheet/src/jupyter/api.tsx:512]: Remove get pivot params?

File: mitosheet/src/hooks/useLiveUpdatingParams.tsx [hunk 6]
```diff
@@ -24,25 +24,39 @@ import { useEffectOnUpdateEvent } from "./useEffectOnUpdateEvent";
...
     analysisData: AnalysisData,
-    debounceDelay: number
+    debounceDelay: number,
+    frontendToBackendConverters?: {
+        getBackendFromFrontend: (params: FrontendParamType) => BackendParamType,
+        getFrontendFromBackend: (params: BackendParamType) => FrontendParamType,
+    },
+    options?: {
+        doNotSendDefaultParams: boolean,
+        executionDataToMatch?: Record<string, string | number>
```
Comment (naterush) [inline mitosheet/src/hooks/useLiveUpdatingParams.tsx:39]: Do I need this? I think no....

**Review by naterush — COMMENTED** [review]
Cleanup and test!

**Comment by aarondr77** [comment]
LGTM once tests pass

**Pull request merged**